### PR TITLE
Update go version to 1.21 as 1.18 is causing issue in AppendUint32

### PR DESCRIPTION
**Pull request by sagar2395**
Issue #90  - Updated go version to 1.21.3 as 1.18 is causing issue in AppendUint32.…
Also fixed lint issues by updating required packages

### CODE REVIEW

**Review by edoardottt — CHANGES_REQUESTED** [review]
golangci-lint is failing, *maybe* go1.21 is the correct format

**Comment by sagar2395** [comment]
@edoardottt Updated to 1.21 version. Let's try using this version. Otherwise, I found another post (https://github.com/golang/go/issues/61851) in which it is mentioned that 1.20 worked fine. If this fails, I can update to that.

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
     steps:
       - uses: actions/setup-go@v3
         with:
-          go-version: 1.17
+          go-version: 1.19
```
Comment (edoardottt) [inline .github/workflows/golangci-lint.yml:21]: 1.21 isn't supported by golangci-lint ?

**Review by edoardottt — COMMENTED** [review]
multiple different Go versions

File: .github/workflows/golangci-lint.yml (hunk 1 above)
Comment (sagar2395) [reply .github/workflows/golangci-lint.yml]: Updated golangci-lint version to 1.21. This should work

**Comment by sagar2395** [comment]
@edoardottt version updated for golangci-lint.

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ jobs:
     steps:
       - uses: actions/setup-go@v3
         with:
-          go-version: 1.17
+          go-version: 1.21
```
Comment (edoardottt) [inline .github/workflows/golangci-lint.yml:21]: golangci-lint is failing, I guess is related to this https://github.com/golangci/golangci-lint/issues/3933#issuecomment-1674808789

**Review by edoardottt — CHANGES_REQUESTED** [review]

File: .github/workflows/golangci-lint.yml (hunk 2 above)
Comment (sagar2395) [reply .github/workflows/golangci-lint.yml:21]: Updated golangci-lint to 1.54.2. This seems to work as per https://github.com/golangci/golangci-lint/issues/3107#issuecomment-1744825702

**Comment by sagar2395** [comment]
Updated golangci-lint to 1.54.2. This seems to work as per https://github.com/golangci/golangci-lint/issues/3107#issuecomment-1744825702

**Review by edoardottt — APPROVED** [review]
lgtm

**Pull request merged**